lib/config.js

```diff
@@ -121,9 +244,13 @@ async function setupTestChain() {
  */
 function dedupeTapos() {
   return {
-    blocksBehind: 3,
+    blocksBehind: TAPOS_BLOCKS_BEHIND,
     // Why the random stuff? To weakly deduplicate repeated transactions
-    expireSeconds: 300 + Math.floor(Math.random() * 3300),
+    // lastTimeAdded compensates for the most recent time adjustment (via addTIme) so that tx expiry is within it
+    expireSeconds:
+      lastTimeAdded +
```

Comment:
is it necessary  to add `lastTimeAdded` , I tried to `addTime` more than 3600 second and the next transaction get error `Transaction expiration is too far in the future relative to the reference time of...`

Reply:
Hmmmm. It is funky but I found it was needed to make the tapos work. I wonder if maybe this is a local timezone issue betwen us. Do the unit tests work for you? In particular, this test https://github.com/worldwide-asset-exchange/waxunit/blob/adjust-time-2/tests/wax-unit.test.js#L335 is meant to check that we are able to go beyond 3600 secods into the future. And how did you check this please? It sounds like this can't be a docker image issue. I would expect the change time call to not work at all in that case. Regardless. You should confirm you have something similar to this ```
aaron@raven:~/dev/projects/wax/wax-unit$ docker images | grep wax-all
731278070712.dkr.ecr.us-east-2.amazonaws.com/wax-all                     latest                    5ebb712f8923        2 days ago          6.44GB
731278070712.dkr.ecr.us-east-2.amazonaws.com/wax-all                     v3.8.0                    5ebb712f8923        2 days ago          6.44GB
```

Reply:
It works well today, I think there is a case that can be happened as the following:

block 1 -> 2 -> 3 -> 4 -> 5 ----> add time more than 3600 second -> 7 -> 8 -> 9 -> 10 -> 11

Assume that user add time from block 5, more than 3600 second -> `lastTimeAdded > 3600`
Then they do not call transaction immediately, because some reasons the code delay more than 2 second
TAPOS get reference from 3 blocks behind head block 11 -> reference block 8
EOSSJ calculate expire time of transaction -> `block(8).timestamp + lastTimeAdded + random>3600`

Reply:
I tried to add `await sleep(1000);` after https://github.com/worldwide-asset-exchange/waxunit/blob/adjust-time-2/tests/wax-unit.test.js#L335 and error happen:

`Transaction expiration is too far in the future relative to the reference time of 2022-03-15T02:59:21.000, expiration is 2022-03-22T01:55:56 and the maximum transaction lifetime is 3600 seconds`

Reply:
yeah you're right. I made a fix that I think was missing and allowed me to remove the lastTimeAdded value which was super suspicious. I added an extra wait for a couple blocks at the end of addTIme which I think gets us in sync with the correct block time after it has freshly been changed, which makes the math make way more sense. I tried it with your sleep(1000) above and it seems to fix that problem too. Can you please check to see if you see any more issues? Thanks!!